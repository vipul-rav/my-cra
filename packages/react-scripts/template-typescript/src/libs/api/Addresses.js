import { v1Api } from 'Libs/Api/Api';

export async function partialAddressSearch (search) {
  const resp = await v1Api.post(`/api/v1/addresses/search/partial`, { search });
  if (resp.ok) return resp.data;
  else {
    console.error(resp.problem);
    return resp.data;
  }
}

// define a concatenated object of methods here to allow for default and named export use
const exportFunctions = {
  partialAddressSearch
};

export default exportFunctions;
