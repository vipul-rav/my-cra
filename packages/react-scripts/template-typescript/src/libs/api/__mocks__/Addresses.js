const Addresses = jest.genMockFromModule('../Addresses').default;

Addresses.partialAddressSearch = () => {
  return { addressData: [{ line1: '123' }] };
};

export default Addresses;
