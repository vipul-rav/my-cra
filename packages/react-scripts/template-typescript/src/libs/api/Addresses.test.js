import Addresses from './Addresses';

jest.mock('./Addresses');

describe('Address API', () => {
  it('searching should match snapshot', () => {
    const resp = Addresses.partialAddressSearch('7 leny road');
    expect(resp).toMatchSnapshot();
  });
});
