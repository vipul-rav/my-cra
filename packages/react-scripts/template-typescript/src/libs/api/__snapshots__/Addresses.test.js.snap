// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Address API searching should match snapshot 1`] = `
Object {
  "addressData": Array [
    Object {
      "line1": "123",
    },
  ],
}
`;
